addons/stock/models/stock_move_line.py

```diff
@@ -321,7 +321,7 @@ def write(self, vals):
         # When editing a done move line, the reserved availability of a potential chained move is impacted. Take care of running again `_action_assign` on the concerned moves.
         if updates or 'qty_done' in vals:
             next_moves = self.env['stock.move']
-            mls = self.filtered(lambda ml: ml.move_id.state == 'done' and ml.product_id.type == 'product')
+            mls = self.filtered(lambda ml: ml.move_id.state == 'done' and ml.product_id.type in ['product', 'consu'])
```

Comment:
In case of consu without a package_id/dest_package_id, you should bypass the _compute_quantity and the _update_available_quantity, no?

Reply:
We also need to capture the case where a package is added to/removed from a consumable's move line. As is this code probably does unnecessary work (does unnecessary quant searches + tries to make a new quant but is ignored in the quant.create when no packages), but the alternative is a messy filter. Maybe something like:

```suggestion
if 'result_package_id' in updates or 'package_id' in updates:
    mls = self.filtered(lambda ml: ml.move_id.state == 'done' and ml.product_id.type in ['product', 'consu'])
else:
    mls = self.filtered(lambda ml: ml.move_id.state == 'done' and (ml.product_id.type == ['product'] or ml.package_id or ml.result_package_id))```

Reply:
You alright, but the condition ` ml.product_id.type in ['product', 'consu']` should be always true because the a move can contains only consu or product. Maybe it is easier to extract the filter in for loop:
```python
for ml in self:
                # Only take reserved ones
                if ml.state not in ('partially_available', 'assigned'):
                    continue
                # By pass consu type expected where there is a package
                if ml.product_id.type == 'consu' and not (ml.package_id or ml.result_package_id):
                    continue
```